experiment/telegram/telegram_otherwise.go

```diff
@@ -0,0 +1,184 @@
+// +build !cgo
+
+// Package telegram contains the Telegram network experiment. This file
+// in particular is a pure-Go implementation of that.
+//
+// See https://github.com/ooni/spec/blob/master/nettests/ts-020-telegram.md.
+package telegram
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ooni/netx/handlers"
+	"github.com/ooni/netx/httpx"
+	"github.com/ooni/probe-engine/experiment"
+	"github.com/ooni/probe-engine/experiment/handler"
+	"github.com/ooni/probe-engine/log"
+	"github.com/ooni/probe-engine/model"
+	"github.com/ooni/probe-engine/session"
+)
+
+const (
+	testName    = "telegram"
+	testVersion = "0.0.2"
+)
+
+// Config contains the experiment config.
+type Config struct{}
+
+// TestKeys contains telegram test keys.
+type TestKeys struct {
+	TelegramHTTPBlocking bool   `json:"telegram_http_blocking"`
+	TelegramTCPBlocking  bool   `json:"telegram_tcp_blocking"`
+	TelegramWebFailure   string `json:"telegram_web_failure"`
```

Comment:
It should be noted that we are changing the semantics of this key. In the current implementation it is `null` when there is no failure, while now it's an empty string.

If you decide to change it be sure to also update the test specification.

Reply:
Good catch, this is a mistake. I'd rather not change the spec.

Reply:
done!